### DISCUSSION TITLE
posts/2021/02/06/%E8%AE%B0%E4%B8%80%E6%AC%A1-nginx-ingress-%E9%95%BF%E8%BF%9E%E6%8E%A5%E5%BC%82%E5%B8%B8%E6%96%AD%E5%BC%80%E9%97%AE%E9%A2%98%E5%A4%8D%E7%9B%98%E8%BF%87%E7%A8%8B

### DISCUSSION THREAD

**Discussion by giscus** [shabbywu/blog#62] [General]
# posts/2021/02/06/%E8%AE%B0%E4%B8%80%E6%AC%A1-nginx-ingress-%E9%95%BF%E8%BF%9E%E6%8E%A5%E5%BC%82%E5%B8%B8%E6%96%AD%E5%BC%80%E9%97%AE%E9%A2%98%E5%A4%8D%E7%9B%98%E8%BF%87%E7%A8%8B

TLDR nginx reload 导致 worker 重启, 而 worker_shutdown_timeout 默认值是 10s (由 nginx-ingress-controller 配置), 导致出现长连接异常断开的问题。 问题背景 最近接到用户反馈, 在 IDC 通过域名访问应用接口时, 当请求耗时到 3 分钟的时候会出现服务端不返回数据直接...

https://blog.shabbywu.cn/posts/2021/02/06/%E8%AE%B0%E4%B8%80%E6%AC%A1-nginx-ingress-%E9%95%BF%E8%BF%9E%E6%8E%A5%E5%BC%82%E5%B8%B8%E6%96%AD%E5%BC%80%E9%97%AE%E9%A2%98%E5%A4%8D%E7%9B%98%E8%BF%87%E7%A8%8B.html

**Comment by pcaqiu** [comment]
你好，请问有定位到nginx reload的原因吗，我也遇到因为nginx reload导致长链接断开的问题。

**Reply by shabbywu** [reply]
时间隔太久了，如果没记错的话，k8s nginx ingress 在扫描到新的 ingress 配置后会做 reload. 主要是 ingress 的域名配置这些在 nginx 这个架构上就是要 reload 才能加载到。（对应 nginx server 这个 block 的配置)

长链接中断问题理论上在 nginx-ingress 是无解的了(除非支持动态配置更新了)，可以延长 worker_shutdown_timeout 来缓解。
